src/pytest_bdd/parser.py

```diff
@@ -413,7 +416,7 @@ def parse_scenario(self, scenario_data: GherkinScenario, feature: Feature) -> Sc
                     for row in example_data.table_body:
                         values = [cell.value or "" for cell in row.cells]
                         examples.add_example(values)
-                    scenario.examples = examples
+                scenario.examples.append(examples)
```

Comment:
`ScenarioTemplate.examples` can be None, weird that mypy didn't catch this

Reply:
this has still to be fixed

Reply:
can we make `ScenarioTemplate.examples` not optional maybe? Or do we need to distinguish between no examples at all and empty examples table?

Reply:
Well, according to the official parser, a scenario is a scenario is a scenario.  All scenarios have an empty list of examples, but if the list is not empty, then it's a scenario outline.

Reply:
I think there was a problem with copy paste in your last comment :)

Reply:
So it looks like we can make it a non optional field, shall we do that ?

Reply:
The gherkin parser logic (which I have been conservative about modifying) used to use this None-ness I guess to distinguish between the two.  I am tempted to go the Gherkin view, which is why I originally modified the type to no longer be:

```
    examples: list[Examples] | None = field(default_factory=list[Examples])
```

but rather

```
    examples: list[Examples] = field(default_factory=list[Examples])
```

I added back the concept of optional based on your original comment.

Reply:
Oh I didn't see you made it non nullable. That's why mypy didn't complain then. Yes please let's have it non nullable then